src/main/java/com/streamr/client/Subscription.java

```diff
@@ -82,20 +77,27 @@ public void setResending(boolean resending) {
         this.resending = resending;
     }
 
-    public ResendRangeRequest handleMessage(StreamMessage msg, String sessionToken) {
-        return handleMessage(msg, sessionToken, false);
+    public void handleMessage(StreamMessage msg) throws GapDetectedException {
+        handleMessage(msg, false);
     }
 
-    public ResendRangeRequest handleMessage(StreamMessage msg, String sessionToken, boolean isResend) {
+    public void handleMessage(StreamMessage msg, boolean isResend) throws GapDetectedException {
         String key = msg.getPublisherId() + msg.getMsgChainId();
+        try {
+            msg.getContent(); // call to trigger potential IOException
```

Comment:
I'm still wondering why this all is necessary? On line `111` we're updating `lastReceivedMsgRef` in a branch where we know that `!checkForGap(msg.getPreviousMessageRef(), key)` is true (or we are resending).

Also we probably shouldn't swallow the exception, if there's something wrong with the message.

What happens if this code is deleted?